pyvista/core/dataset.py

```diff
@@ -169,32 +257,104 @@ def active_tensors_name(self) -> str:
 
     @active_tensors_name.setter
     def active_tensors_name(self, name: str):
-        """Set the name of the active tensor."""
         self.set_active_tensors(name)
 
     @property
     def active_vectors_name(self) -> str:
-        """Return the name of the active vectors array."""
+        """Return the name of the active vectors array.
+
+        Examples
+        --------
+        Create a mesh, compute the normals, set them as active, and
+        return the name of the active vectors.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Sphere()
+        >>> mesh_w_normals = mesh.compute_normals()
+        >>> mesh_w_normals.active_vectors_name = 'Normals'
+        >>> mesh_w_normals.active_vectors_name
+        'Normals'
+
+        """
         return self.active_vectors_info.name
 
     @active_vectors_name.setter
     def active_vectors_name(self, name: str):
-        """Set the name of the active vector."""
         self.set_active_vectors(name)
 
-    @property
-    def active_scalars_name(self) -> str:
-        """Return the active scalar's name."""
+    @property  # type: ignore
+    def active_scalars_name(self) -> str:  # type: ignore
```

Comment:
I don't know anything about typing, but do we really need to place an ignore on both lines? And shouldn't we use `Optional[str]` instead of an ignore?

Reply:
I spent a good 30 minutes learning about the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) and why `mypy` is flagging this (no matter what I do).  Issue is that this property is expected in `DataSetFilters` since according to the principle you should be able to substitute the use of one class for another.  Since `DataSetFilters` is also an abstract class, this shouldn't apply.

`mypy` either doesn't have hooks for abstract classes or I'm not knowledgeable enough to use them.

Reply:
OK, as long as it's on purpose :) Thanks. I wouldn't be surprised if whatever mypy had in the way of abstract base classes would only work with stdlib machinery. Then again I'd expect that abstractness shouldn't matter for typing.